### PR TITLE
refactor: Refactor session storage to decouple oidc from non-oidc

### PR DESCRIPTION
**Pull request by zredinger-ccc**

### CODE REVIEW

File: dbtype/types.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+type Session struct {
+	ID        ccc.UUID  `spanner:"Id"        db:"Id"`
+	Username  string    `spanner:"Username"  db:"Username"`
+	CreatedAt time.Time `spanner:"CreatedAt" db:"CreatedAt"`
+	UpdatedAt time.Time `spanner:"UpdatedAt" db:"UpdatedAt"`
+	Expired   bool      `spanner:"Expired"   db:"Expired"`
+}
+
+type SessionOIDC struct {
```
Comment (hdishner-ccc) [inline dbtype/types.go:18]: Would embedding the default Session and InsertSession structs make sense here?

File: spanner_preauth.go [hunk 2]
```diff
@@ -41,3 +43,46 @@ func (p *SpannerPreauthSessionStorage) NewSession(ctx context.Context, username
 
 	return id, nil
 }
+
+// Session returns the session information from the database for given sessionID
+func (p *SpannerPreauthSessionStorage) Session(ctx context.Context, sessionID ccc.UUID) (*sessioninfo.SessionInfo, error) {
+	ctx, span := otel.Tracer(name).Start(ctx, "Client.Session()")
```
Comment (hdishner-ccc) [inline spanner_preauth.go:49]:
```suggestion
	ctx, span := otel.Tracer(name).Start(ctx, "SpannerPreauthSessionStorage.Session()")
```

File: postgres_preauth.go [hunk 3]
```diff
@@ -40,3 +42,46 @@ func (p *PostgresPreauthSessionStorage) NewSession(ctx context.Context, username
 
 	return id, nil
 }
+
+// Session returns the session information from the database for given sessionID
+func (p *PostgresPreauthSessionStorage) Session(ctx context.Context, sessionID ccc.UUID) (*sessioninfo.SessionInfo, error) {
+	ctx, span := otel.Tracer(name).Start(ctx, "Client.Session()")
```
Comment (hdishner-ccc) [inline postgres_preauth.go:48]:
```suggestion
	ctx, span := otel.Tracer(name).Start(ctx, "PostgresPreauthSessionStorage.Session()")
```

File: dbtype/types.go (hunk 1 above)
Comment (zredinger-ccc) [reply dbtype/types.go:18]: Just pushed a tweak to that. LMK your thoughts.

**Review by hdishner-ccc — APPROVED** [review]
LGTM

**Pull request merged**